src/main/java/org/broadinstitute/hellbender/tools/exome/germlinehmm/IntegerCopyNumberTransitionMatrixCollection.java

```diff
@@ -0,0 +1,167 @@
+package org.broadinstitute.hellbender.tools.exome.germlinehmm;
+
+import org.apache.commons.math3.util.FastMath;
+import org.broadinstitute.hellbender.exceptions.UserException;
+
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.io.Reader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class holds a collection of {@link IntegerCopyNumberTransitionMatrixData} for different sex genotypes
+ * and contigs.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class IntegerCopyNumberTransitionMatrixCollection {
+
+    /**
+     * The map {@link SexGenotypeContigPairKey} => integer copy number prior data
+     */
+    private final Map<SexGenotypeContigPairKey, IntegerCopyNumberTransitionMatrixData>
+            sexGenotypeAndContigToTransitionMatrixMap;
+
+    private int maxCopyNumberInCollection;
+
+    /**
+     *
+     */
+    public IntegerCopyNumberTransitionMatrixCollection() {
+        sexGenotypeAndContigToTransitionMatrixMap = new HashMap<>();
+        maxCopyNumberInCollection = -1;
+    }
+
+    /**
+     *
+     * @param sexGenotype
+     * @param contig
+     * @param priorData
+     */
+    public void add(@Nonnull final String sexGenotype,
+                    @Nonnull final String contig,
+                    @Nonnull final IntegerCopyNumberTransitionMatrixData priorData) {
+        final SexGenotypeContigPairKey key = SexGenotypeContigPairKey.of(sexGenotype, contig);
+        if (sexGenotypeAndContigToTransitionMatrixMap.keySet().contains(key)) {
+            throw new IllegalArgumentException(String.format("The integer copy number transition matrix for sex" +
+                    " genotype \"%s\" and contig \"%s\" is already defined. Check for duplicate entries in the" +
+                    " table file.", sexGenotype, contig));
+        }
+        sexGenotypeAndContigToTransitionMatrixMap.put(key, priorData);
+        maxCopyNumberInCollection = FastMath.max(maxCopyNumberInCollection, priorData.getMaxCopyNumber());
+    }
+
+    /**
+     *
+     * @param sexGenotype
+     * @param contig
+     * @return
+     */
+    public IntegerCopyNumberTransitionMatrixData get(@Nonnull final String sexGenotype,
+                                                     @Nonnull final String contig) {
+        return sexGenotypeAndContigToTransitionMatrixMap.get(checkKey(sexGenotype, contig));
+    }
+
+    /**
+     * Returns a set of sex genotypes that are annotated at least once
```

Comment:
Since the collection is filled in incrementally via `add(...)`, it does not need to have all [sex genotype, contig] annotations in every state, i.e. it can be in an _incomplete_ state. The method `assertCompleteness` checks for that. I added this to the javadoc.